code/modules/antagonists/heretic/knowledge/knock_lore.dm

```diff
@@ -0,0 +1,242 @@
+/**
+ * # The path of Knock.
+ *
+ * Goes as follows:
+ *
+ * A Locksmith’s Secret
+ * Grasp of Knock
+ * > Sidepaths:
+ *   Ashen Eyes
+ *	 Codex Cicatrix
+ * Key Keeper’s Burden
+ *
+ * Rite Of Passage
+ * Mark Of Knock
+ * Ritual of Knowledge
+ * Burglar's Finesse
+ * Apetra Vulnera
+ * > Sidepaths:
+ *   Blood Siphon
+ *   Void Cloak
+ *
+ * Opening Blade
+ * Caretaker’s Last Refuge
+ *
+ * Many secrets behind the Spider Door
+ */
+/datum/heretic_knowledge/limited_amount/starting/base_knock
+	name = "A Locksmith’s Secret"
+	desc = "Opens up the Path of Knock to you. \
+		Allows you to transmute a knife and a crowbar a Key Blade. \
+		You can only create two at a time, and they function as fast crowbars."
+	gain_text = "The Knock permits no seal and no isolation. It thrusts us gleefully out of the safety of ignorance."
+	next_knowledge = list(/datum/heretic_knowledge/knock_grasp)
+	required_atoms = list(
+		/obj/item/knife = 1,
+		/obj/item/crowbar = 1,
+	)
+	result_atoms = list(/obj/item/melee/sickly_blade/knock)
+	limit = 2
+	route = PATH_KNOCK
+
+/datum/heretic_knowledge/knock_grasp
+	name = "Grasp of Knock"
+	desc = "Your mansus grasp allows you to access anything! Right click on an airlock or a locker to force it open. \
+		DNA locked mechs will remove the lock and force the pilot out. Works on consoles. \
+		Makes a distinctive knocking sound on use."
+	gain_text = "My new found desires drove me to greater and greater heights."
+	next_knowledge = list(
+	/datum/heretic_knowledge/key_ring,
+	/datum/heretic_knowledge/medallion,
+	/datum/heretic_knowledge/codex_cicatrix,
+	)
+	cost = 1
+	route = PATH_KNOCK
+
+/datum/heretic_knowledge/knock_grasp/on_gain(mob/user, datum/antagonist/heretic/our_heretic)
+	RegisterSignal(user, COMSIG_HERETIC_MANSUS_GRASP_ATTACK_SECONDARY, PROC_REF(on_secondary_mansus_grasp))
+	RegisterSignal(user, COMSIG_HERETIC_MANSUS_GRASP_ATTACK, PROC_REF(on_mansus_grasp))
+
+/datum/heretic_knowledge/knock_grasp/on_lose(mob/user, datum/antagonist/heretic/our_heretic)
+	UnregisterSignal(user, COMSIG_HERETIC_MANSUS_GRASP_ATTACK_SECONDARY)
+	UnregisterSignal(user, COMSIG_HERETIC_MANSUS_GRASP_ATTACK)
+
+/datum/heretic_knowledge/knock_grasp/proc/on_mansus_grasp(mob/living/source, mob/living/target)
+	SIGNAL_HANDLER
+	var/obj/item/clothing/under/suit = target.get_item_by_slot(ITEM_SLOT_ICLOTHING)
+	if(istype(suit) && suit.adjusted == NORMAL_STYLE)
+		suit.toggle_jumpsuit_adjust()
+		suit.update_appearance()
+
+/datum/heretic_knowledge/knock_grasp/proc/on_secondary_mansus_grasp(mob/living/source, atom/target)
+	SIGNAL_HANDLER
+	
+	if(ismecha(target))
+		var/obj/vehicle/sealed/mecha/mecha = target
+		mecha.dna_lock = null
+		for(var/mob/living/occupant as anything in mecha.occupants)
+			if(isAI(occupant))
+				continue
+			mecha.mob_exit(occupant, randomstep = TRUE)
+	else if(istype(target,/obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/door = target
+		door.unbolt()
```

Comment:
AFAIK knocked doors (ie, `COMSIG_ATOM_MAGICALLY_UNLOCKED`) unbolt themselves anyways so this isn't necessary?

Reply:
they dont do this

Reply:
i think adding it to the signal on airlocks would be a balance pr